src/object/schedcontrol.c

```diff
@@ -28,9 +28,11 @@ static exception_t invokeSchedControl_Configure(sched_context_t *target, word_t
 #ifdef ENABLE_SMP_SUPPORT
             if (target->scCore == getCurrentCPUIndex()) {
 #endif /* ENABLE_SMP_SUPPORT */
-                if (checkBudget()) {
-                    commitTime();
-                }
+                /* This could potentially mutate state but if it returns
+                 * true no state was modified, thus removing it should
+                 * be the same. */
+                assert(checkBudget());
```

Comment:
Is this in an assert because the if the current SC is performing this invocation then checkBudget must have returned true upon entry to the kernel?

Reply:
Yes